common/unsafe/src/test/java/org/apache/spark/unsafe/types/UTF8StringSuite.java

```diff
@@ -362,15 +362,15 @@ public void pad() {
 
     assertEquals(EMPTY_UTF8, fromString("数据砖头").lpad(-10, fromString("孙行者")));
     assertEquals(EMPTY_UTF8, fromString("数据砖头").lpad(-10, EMPTY_UTF8));
-    assertEquals(fromString("数据砖头"), fromString("数据砖头").lpad(5, EMPTY_UTF8));
-    assertEquals(fromString("数据砖"), fromString("数据砖头").lpad(3, EMPTY_UTF8));
-    assertEquals(EMPTY_UTF8, EMPTY_UTF8.lpad(3, EMPTY_UTF8));
+    assertEquals(fromString(null), fromString("数据砖头").lpad(5, EMPTY_UTF8));
+    assertEquals(fromString(null), fromString("数据砖头").lpad(3, EMPTY_UTF8));
+    assertEquals(fromString(null), EMPTY_UTF8.lpad(3, EMPTY_UTF8));
 
     assertEquals(EMPTY_UTF8, fromString("数据砖头").rpad(-10, fromString("孙行者")));
     assertEquals(EMPTY_UTF8, fromString("数据砖头").rpad(-10, EMPTY_UTF8));
-    assertEquals(fromString("数据砖头"), fromString("数据砖头").rpad(5, EMPTY_UTF8));
-    assertEquals(fromString("数据砖"), fromString("数据砖头").rpad(3, EMPTY_UTF8));
-    assertEquals(EMPTY_UTF8, EMPTY_UTF8.rpad(3, EMPTY_UTF8));
+    assertEquals(fromString(null), fromString("数据砖头").rpad(5, EMPTY_UTF8));
```

Comment:
So, the previous behaivour was like PostgreSQL before but you propose to match it to Hive, Mysql and Oracle?

Reply:
If that's the case, I am less sure why we should necessarily change. The lpad and rpad implementations seem different per DBMS implementation. Spark's case at least has one reference and the current behaviour makes sense as well.

Reply:
It is reasonable to change this because as per description of this function 
In case of empty pad string, the return value should be null.

Reply:
@HyukjinKwon, I feel reasonable because in other DBMS like hive, Mysql they handle this issue after they find this problem.

Reply:
This function behaviour varies per DBMS implementations and we have a reference as you described, PostgreSQL. Can you guys elaborate why it looks reasonable to you guys? I don't see a strong reason to change the current behaviour.

Reply:
I think NULL is the better choice, that way we don't return unpadded data.
NULL makes sense with invalid input.
Please refer the JIRA how Hive handled
https://issues.apache.org/jira/browse/HIVE-15792
So as per me we should follow this as per Hive.